nixos/modules/services/networking/wg-access-server.nix

```diff
@@ -0,0 +1,118 @@
+{ config, pkgs, lib, ... }:
+let
+  inherit (lib) mkEnableOption mkPackageOption mkOption types;
+
+  cfg = config.services.wg-access-server;
+
+  settingsFormat = pkgs.formats.yaml { };
+  configFile = settingsFormat.generate "config.yaml" cfg.settings;
+in
+{
+
+  options.services.wg-access-server = {
+    enable = mkEnableOption "wg-access-server";
+
+    package = mkPackageOption pkgs "wg-access-server" { };
+
+    settings = mkOption {
+      type = settingsFormat.type;
```

Comment:
Should be a freeformType, something along the lines of

```nix
settings = mkOption {
  type = lib.types.submodule {
    freeformType = settingsFormat.type;
    options.dns.enabled = mkOption {
      type = types.bool;
      default = true;
    };
    options.storage = mkOption {
      type = types.str;
      default = "sqlite3://db.sqlite";
    };
  };
};
```

Which gives us proper documentation and type checking for known values.